pkg/kubelet/apis/deviceplugin/v1beta1/api.proto

```diff
@@ -48,6 +55,11 @@ service DevicePlugin {
 	// Plugin can run device specific operations and instruct Kubelet
 	// of the steps to make the Device available in the container
 	rpc Allocate(AllocateRequest) returns (AllocateResponse) {}
+
+        // PreStart is called, if indicated by Device Plugin during registeration phase,
+        // before each container start. Device plugin can run device specific operations
+        // such as reseting the device before making devices available to the container
+	rpc PreStart(PreStartRequest) returns (PreStartResponse) {}
```

Comment:
nit: Can you name it `PreStartContainer` coz we may have a `PreStartPod` in the future (potentially for Network Plugins)?